### PR TITLE
fix: Refactor buyer name fields in order-form

### LINKED ISSUE #2707: Username changing along with Ticket Buyer Name
**Describe the bug**

When ordering a ticket, the ticket buyer name is changing along with the user name on top

**To Reproduce**
ezgif com-video-to-gif (2)

**Expected behaviour**

The User name shouldn't be changing at all

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Ubuntu16.04
 - Browser [e.g. chrome, safari] Chrome

**Additional context**

Working on it

### PR DESCRIPTION
**Pull request by uds5501**
- This avoid name changes in the nav bar because the name has been pre loaded

Fixes #2707

#### Short description of what this resolves:
This utilises computed property to make sure that user name doesn't change during filling of forms.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by uds5501** [comment]
@shreyanshdwivedi @kushthedude @mrsaicharan1 Peer review time!! :smile:

**Comment by uds5501** [comment]
@CosmicCoder96 in that case, this issue shouldn't be valid then, if I am understanding you?

**Comment by CosmicCoder96** [comment]
@uds5501 No this issue is valid, the buyer name changing the username is wrong.

**Comment by uds5501** [comment]
@CosmicCoder96 I think I maybe incorrect in the interpretation of the term *sync* but you mean that it should change and correctly signify the users : 

- When a new user is logged in
- When a profile name is updated
- User logs out 

Correct?
If that is the case then this implementation works locally. Although if there is some other case which I might've not covered, please let me know

File: app/components/nav-bar.js [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
 
 export default Component.extend({
   actions: {
     logout() {
       this.authManager.logout();
       this.routing.transitionTo('index');
     }
-  }
+  },
+  nameShown: computed('authManager.currentUser', function() {
```
Comment (niranjan94) [inline app/components/nav-bar.js]: Not a valid fix. This should be corrected at the source of the problem which is the order form. 

`authManager.currentUser` should be considered the source of truth for the current logged in user. the navbar and all other UI elements should always reflect what it contains. 

but the order form on the other hand, _must not_ modify `authManager.currentUser` directly when a user input is still happening. update of `authManager.currentUser` should only happen when the the data on the server itself changes. (ie either buy an update request to the server modifying the user data or after a GET request retrieving the latest user data)

Comment (uds5501) [reply app/components/nav-bar.js]: @niranjan94 Understood, I will try to formulate another approach . I think I can utilise the `buyer` attribute in the order-form component

**Comment by uds5501** [comment]
This is working for me in my local, @shreyanshdwivedi @kushthedude and @mrsaicharan1 could you please verify it ? If it's working fine, then @CosmicCoder96 please review!

File: app/components/forms/orders/order-form.js [hunk 2]
```diff
@@ -14,14 +14,26 @@ import {
 export default Component.extend(FormMixin, {
   router: service(),
 
+  buyerFirstName: computed('buyerHasFirstName', function() {
```
Comment (niranjan94) [inline app/components/forms/orders/order-form.js]: Never mutate a plain computed property with only a getter directly. That is an incorrect usage. Use [`oneWay`](https://api.emberjs.com/ember/3.10/functions/@ember%2Fobject%2Fcomputed/oneWay) here instead. That exists exactly for this purpose.

File: app/components/forms/orders/order-form.js [hunk 3]
```diff
@@ -14,14 +14,26 @@ import {
...
+    if (this.buyerHasFirstName) {
+      return this.get('data.user.firstName');
+    } else {
+      return '';
+    }
+  }),
+  buyerLastName: computed('buyerHasLastName', function() {
```
Comment (niranjan94) [inline app/components/forms/orders/order-form.js]: See comment above

File: app/components/forms/orders/order-form.js [hunk 4]
```diff
@@ -14,14 +14,26 @@ import {
...
+    if (this.buyerHasLastName) {
+      return this.get('data.user.lastName');
+    } else {
+      return '';
+    }
+  }),
   buyer: computed('data.user', function() {
```
[the remark this answers is not in the document]
Comment (uds5501) [reply app/components/forms/orders/order-form.js]: @niranjan94 sure, in this ticket I will handle this file particularly, I shall open another issue for such cases once this is done

File: app/components/forms/orders/order-form.js (hunk 2 above)
Comment (uds5501) [reply app/components/forms/orders/order-form.js]: @niranjan94 If I am understanding your concern correctly, I should use it in this way, right?
```js
buyerFirstName: oneway('buyerHasFirstName')
```

Comment (kushthedude) [reply app/components/forms/orders/order-form.js]: @uds5501 No ,
```
 buyerFirstName: computed.oneWay('buyerHasFirstName')
```

**Comment by uds5501** [comment]
@kushthedude @niranjan94 please have a look now

**Comment by uds5501** [comment]
@CosmicCoder96 Please review

**Comment by mrsaicharan1** [comment]
@uds5501 Is this working locally? I still see that the name changes in netlify deployment. Can you please check it out? Would be great if you could confirm.

**Comment by uds5501** [comment]
@mrsaicharan1 weird, let me have a look.

**Comment by uds5501** [comment]
@mrsaicharan1 I tried ordering a ticket without logging in. Looks like it's working fine. Can you please pinpoint where is it going wrong?
image
image

**Comment by mrsaicharan1** [comment]
@uds5501 Try logging in and changing the first name and last name while ordering. navbar name changes.

**Comment by mrsaicharan1** [comment]
May-30-2019 13-35-33

**Comment by uds5501** [comment]
@mrsaicharan1 PHEW. Yes, that's something which it's supposed to do. remember this PR? #2948.
This issue pertains to the fact that while typing in "Buyer First Name", the navbar was changing along with it. And from your GIF, this is clear that it works as intended

**Comment by mrsaicharan1** [comment]
> @mrsaicharan1 PHEW. Yes, that's something which it's supposed to do. remember this PR? #2948.
> This issue pertains to the fact that while typing in "Buyer First Name", the navbar was changing along with it. And from your GIF, this is clear that it works as intended

But once you log back in, it still remains. Is that supposed to happen? If it is, then it's fine!

**Comment by uds5501** [comment]
@mrsaicharan1 Yes, it does. Actually, if a user has no name and he's filling up the buyer form, the user name gets updated and hence the navbar is updated as well.

**Comment by uds5501** [comment]
@niranjan94 please review

**Comment by uds5501** [comment]
@CosmicCoder96 @niranjan94 @SaptakS 
The related issues has been added to this week's milestone, please take a look and if there is anything to change, let me know!

**Comment by uds5501** [comment]
@niranjan94 Please review!

**Pull request merged**